README.md

```diff
@@ -1,8 +1,65 @@
 # Host Socket Sharing
 
-This repository contains the HSS specification as well as Xaptums host-side implementation. 
+This repository contains the HSS specification as well as Xaptums implementation for both hosts and devices.
 
-See `HSS_protocol.pdf` for more information.
+See `HSS_protocol.pdf` for information regarding the HSS standard.
+
+# HSS Host Driver
+
+The `host` directory contains the driver required for an HSS host to sit between the USB device and the external network.
+
+## Debian Installation
+
+Xaptum maintains an Apt repository for HSS, so all Debian users need to do is add Xaptums repo and install.
+
+```
+apt-key adv --keyserver keyserver.ubuntu.com --recv-keys c615bfaa7fe1b4ca
+echo "deb http://dl.bintray.com/xaptum/deb <dist> main" | sudo tee /etc/apt/sources.list.d/xaptum.list
+sudo apt-get update
+sudo apt-get install xaptum-hss-host
+```
+Replace <dist> with your Debian distrobuation (Stretch, Buster, ect)
+
+
+
+## DKMS install
+
+The preferred installation method for non-Debian users is using DKMS, otherwise this driver will have to be manually rebuild every time the kernel is changed. A script is provided in the host directory to automate the DKMS install process.
+
+```
+git clone git@github.com:xaptum/hss.git
+cd hss/host
+sudo ./dkms-install
+```
+
+## Manual installation
+
+If neither our APT repos or DKMS are available the module can be independently built and managed.
+
+Note: Required packages: kernel-devel, git, build-essential, kernel-headers-$(uname -r), sudo
+
+```
+git clone git@github.com:xaptum/hss.git
+cd hss/host/src
+make
+sudo make install
+```
+
+# HSS Device Drivers
+
+HSS requires out-of-tree modifications to the kernel of the USB device, as such patches must be applied to the kernel as it is build. Luckily, the patches are relatively simple. They are located in the `device/kernel` directory. The patch `hss-includes.patch` creates new includes to device HSS packet specifications within the kernel, `hss-socket.patch` creates a new type of socket `AF_HSS`, and `hss-selinux.patch` adds this new sockets definitions to SELinux's compile time checks. The second two patches may require some amount of customization to fit around a given version of Linux, but the addition is very simple.
```

Comment:
... creates new `include` statements in the device HSS...

also - break up long lines.